redfish/softwareinventory.go

```diff
@@ -0,0 +1,62 @@
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+package redfish
+
+import (
+	"encoding/json"
+
+	"github.com/stmcginnis/gofish/common"
+)
+
+// SoftwareInventory is used to represent a software or firmware item on the server
+type SoftwareInventory struct {
+	common.Entity
+
+	Description            string
+	Status                 common.Status
+	Version                string
+	LowestSupportedVersion string
+	Manufacturer           string
+	ReleaseDate            string
+	SoftwareID             string
+	UefiDevicePaths        []string
+	Updateable             bool
+	WriteProtected         bool
+	rawData                []byte
```

Comment:
Since this object doesn't support setting values via the Update() call, the capture of the rawData could be removed.